game/nintendo_mii/gen2_wiiu_3ds_miitomo_mii.ksy

```diff
@@ -0,0 +1,244 @@
+meta:
+  id: gen2_wiiu_3ds_miitomo_mii
+  title: Wii U, 3DS, Miitomo Mii character data
+  file-extension:
+    - cfsd
+    - ffsd
+  endian: le
+seq:
+  -
+    type: u1
+    doc: |
+      Currently unknown data.
+      Always 3?
+      May be an internal version identifier?
+  - id: character_set
+    type: b2
+    enum: character_sets
+  - id: region_lock
+    type: b2
+    enum: region_locks
+  - id: is_profanity_flag_enabled
+    type: b1
+  - id: is_copying_enabled
+    type: b1
+  -
+    type: b2
+    doc: Currently unknown data.
+  - id: mii_position_slot_index
+    type: b4
+  - id: mii_position_page_index
+    type: b4
+  - id: version
+    type: b4
+  -
+    type: b4
+    doc: Currently unknown data.
+  - id: system_id
+    type: u1
+    repeat: expr
+    repeat-expr: 8
+  - id: avatar_id
+    type: u1
+    repeat: expr
+    repeat-expr: 4
+  - id: client_id
+    type: u1
+    repeat: expr
+    repeat-expr: 6
+  - id: padding
+    type: u2
+  - id: data_1
+    type: u2
+  - id: mii_name
+    type: str
+    size: 20
+    encoding: utf-16le
+  - id: body_height
+    type: u1
+  - id: body_weight
+    type: u1
+  - id: face_color
+    type: b3
+  - id: face_type
+    type: b4
+  - id: mingle
+    type: b1
+    doc: More info needed. Does this mean IS Mingled or IS NOT Mingled, such as on the Wii?
+  - id: face_makeup
+    type: b4
+  - id: face_wrinkles
+    type: b4
+  - id: hair_type
+    type: u1
+  -
+    type: b4
+    doc: Currently unknown data.
+  - id: is_hair_flipped
+    type: b1
+  - id: hair_color
+    type: b3
+  - id: eye
+    type: u4
+  - id: eyebrow
+    type: u4
+  - id: nose
+    type: u2
+  - id: mouth
+    type: u2
+  - id: mouth2
+    type: u2
+  - id: beard
+    type: u2
+  - id: glasses
+    type: u2
+  - id: mole
+    type: u2
+  - id: creator_name
+    type: str
+    size: 20
+    encoding: utf-16le
+  - id: padding2
+    type: u2
+  - id: checksum
+    type: u2
+instances:
+  favorite:
+    value: data_1 >> 14 & 1
+    doc: Whether the Mii is a favorite or not.
+  favorite_color:
+    value: data_1 >> 10 & 15
+    enum: favorite_colors
+    doc: Favorite color. Ranges from 0 to 11.
+  birth_day:
+    value: data_1 >> 5 & 31
+    doc: Mii birthday day, Ranges from 0 to 30
+  birth_month:
+    value: data_1 >> 1 & 15
+    enum: months
+    doc: Mii birthday month, Ranges from 0 to 11
+  gender:
+    value: data_1 & 1
+    enum: genders
+    doc: Mii gender. 0 = male, 1 = female.
```

Comment:
Consider rewriting all such manual bit operations using the [little-endian bit integers](https://doc.kaitai.io/user_guide.html#bit-ints-le). For the `data_1` bitfield, it would mean replacing this

<del>

```ksy
  - id: data_1
    type: u2
```
</del>

with this:

```ksy
meta:
  bit-endian: le
seq:
  # ...
  - id: gender
    type: b1
    enum: genders
    doc: Mii gender
  - id: birth_month
    type: b4
    enum: months
    doc: Mii birthday month, Ranges from 0 to 11
  - id: birth_day
    type: b5
    doc: Mii birthday day, Ranges from 0 to 30
  - id: favorite_color
    type: b4
    enum: favorite_colors
    doc: Favorite color. Ranges from 0 to 11.
  - id: is_favorite
    type: b1
    doc: Whether the Mii is a favorite or not.
  - id: unused # I assume this bit is not unused; if you consider it rather unknown, sure, omit the `id`
    type: b1
```

Read https://doc.kaitai.io/user_guide.html#bit-ints-le for more info.

Reply:
@HEYimHeroic, would this interfere with "Mii Studio Codes" at all?

Reply:
I already did that in the Kaitai that I made. See my latest comment.